### PR TITLE
Updated Jetty dependency and a test broken by Jetty changes.

### PR DESCRIPTION
**Pull request by jakaarl**
See issue: #694

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -30,7 +30,7 @@
 
     <properties>
         <java.version>1.8</java.version>
-        <jetty.version>9.3.6.v20151106</jetty.version>
+        <jetty.version>9.3.13.v20161014</jetty.version>
```
Comment (ph-hs) [inline pom.xml]: 9.4.0.v20161208 is out now.

**Comment by ph-hs** [comment]
9.4.0.v20161208 is out now

**Comment by jakaarl** [comment]
@ph-hs I noticed, started updating the pull request last night, but there were complications.

**Comment by paul-hammant** [comment]
It is interesting that it is only the test that's broken with the Jetty upgrade. In real life, you can do the maven <excludes/> thing and deploy Spark with the 9.4.x of Jetty quite happily.

**Comment by paul-hammant** [comment]
The bits that you're stuck on could be, solved by:

```
-        WebSocketUpgradeFilter webSocketUpgradeFilter =
-                (WebSocketUpgradeFilter) servletContextHandler.getAttribute("org.eclipse.jetty.websocket.server.WebSocketUpgradeFilter");
+        WebSocketUpgradeFilter webSocketUpgradeFilter = (WebSocketUpgradeFilter) servletContextHandler.getServletHandler().getFilter("Jetty_WebSocketUpgradeFilter").getFilter();
```

and 

```
-        WebSocketServerFactory webSocketServerFactory = webSocketUpgradeFilter.getFactory();
+        WebSocketServerFactory webSocketServerFactory = webSocketUpgradeFilter.getConfiguration().getFactory();
```

and 

```
MappedResource<WebSocketCreator> mappedResource = webSocketUpgradeFilter.getMappings().getMatch("/websocket");
+        MappedResource<WebSocketCreator> mappedResource = webSocketUpgradeFilter.getConfiguration().getMatch("/websocket");
```

**Comment by jakaarl** [comment]
Thanks, I'll take a look tonight. I got burned by not getting the upgrade filter from the attributes and didn't have the time/energy to research into it last night.

**Comment by ph-hs** [comment]
One of the unrelated tests is now failing:

```
testDirectoryTraversalProtectionLocal(spark.staticfiles.StaticFilesTest)  Time elapsed: 0.13 sec  <<< FAILURE!
java.lang.AssertionError: expected:<404> but was:<400>
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotEquals(Assert.java:834)
	at org.junit.Assert.assertEquals(Assert.java:645)
	at org.junit.Assert.assertEquals(Assert.java:631)
	at spark.staticfiles.StaticFilesTest.testDirectoryTraversalProtectionLocal(StaticFilesTest.java:145)
```

Needs to be investigated, I think.

**Comment by jakaarl** [comment]
Seems the unexpected response comes all the way from Jetty:
`WARN org.eclipse.jetty.http.HttpParser - bad HTTP parsed: 400 Bad URI for HttpChannelOverHttp@4db7b425{r=1,c=false,a=IDLE,uri=//localhost:4567/..%5Cspark%5CSpark.class}`

**Comment by tipsy** [comment]
@jakaarl would be great if you could have a look at this again.

**Pull request merged**